### PR TITLE
Add TaskDuration

### PR DESCRIPTION
**Pull request by Ch40gRv1-Mu**
Refactor duration to make it have meaning of time. Also supports flexible format of the time.

### CODE REVIEW

File: src/main/java/seedu/duke/data/TaskDuration.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.ParseException;
+import seedu.duke.exceptions.WrongDurationFormatException;
+
+
+public class TaskDuration {
+
+
+    private static final String DURATION_STRING_FORMAT = "(?<duration>[1-9]\\d*\\.?\\d*|0\\.\\d*[1-9])"
```
Comment (heekit73098) [inline src/main/java/seedu/duke/data/TaskDuration.java]: Consider shifting all these to the StringConstants

File: src/main/java/seedu/duke/data/TaskDuration.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        // no legal input unit match, throw exception
+        throw new WrongDurationFormatException();
+
+    }
+
+
+    // Intentionally implement the parse method independently here, because later will refact the command parser.
+    private HashMap<String, String> parseDurationString(String durationString) throws ModHappyException {
```
Comment (heekit73098) [inline src/main/java/seedu/duke/data/TaskDuration.java:76]: i think keeping it here is fine for 2.0 but maybe we can consider shifting this to parser for 2.1 to keep it consistent

File: src/main/java/seedu/duke/util/StringConstants.java [hunk 3]
```diff
@@ -196,6 +197,7 @@
             + "View all available config settings with \"option\".";
     public static final String ERROR_MODULE_LIST_EMPTY = "Sorry, you have 0 MCs counted towards your GPA ._.\n"
             + "Please add some modules or grades!";
+    public static final String ERROR_WRONG_DURATION_FORMAT = "Sorry, the estimated time is in wrong format Lol";
```
Comment (heekit73098) [inline src/main/java/seedu/duke/util/StringConstants.java]: this Lo1 looks like lol instead so im not sure if u want to keep it

**Review by heekit73098 — CHANGES_REQUESTED** [review]
Generally good implementation, just a few minor changes

File: src/main/java/seedu/duke/data/TaskDuration.java (hunk 1 above)
Comment (Ch40gRv1-Mu) [reply src/main/java/seedu/duke/data/TaskDuration.java]: Okk

File: src/main/java/seedu/duke/data/TaskDuration.java (hunk 2 above)
Comment (Ch40gRv1-Mu) [reply src/main/java/seedu/duke/data/TaskDuration.java:76]: Yep, I plan to refactoring the Parser for commands later. Hence don't want this part to be affected later.

File: src/main/java/seedu/duke/util/StringConstants.java (hunk 3 above)
Comment (Ch40gRv1-Mu) [reply src/main/java/seedu/duke/util/StringConstants.java]: okk

**Pull request merged**